test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -213,3 +220,39 @@ def test_execjob_end_non_blocking_multi_node(self):
                 "execjob_end hook executed at: %s,"
                 "exechost_periodic at: %s and execjob_end hook ended at: %s" %
                 (date_time1, date_time2, date_time3))
+
+    def test_execjob_end_delete_request(self):
+        """
+        Test to make sure execjob_end hook is running
+        after job force deletion request(IS_DISCARD_JOB) when
+        mom is unblocked.
+        """
+        hook_name = "execjob_end_logmsg6"
+        self.server.create_import_hook(hook_name, self.attr, self.hook_body)
+        if len(self.moms) > 1:
+            a = {'Resource_List.select': '2:ncpus=1',
+                 'Resource_List.place': "scatter"}
+            j = Job(TEST_USER, attrs=a)
```

Comment:
I suggest optimizing the if-else condition by just initializing job object as that is only different. Rest can be common.

Reply:
Thanks! I've pushed a new commit addressing this.